### ISSUE TITLE
Cannot import all macros from crate

### ISSUE THREAD

**Issue by StragaSevera** [de-vri-es/assert2-rs#19]
```rust
#[cfg(test)]
mod tests {
    use assert2::*;

    #[test]
    fn it_works() {
        check!(2 + 2 >= 5);
    }
}
```
gives: 
```
error: recursion limit reached while expanding `stringify!`
  --> src/lib.rs:32:9
   |
32 |         check!(2 + 2 >= 5);
   |         ^^^^^^^^^^^^^^^^^^^
   |
   = help: consider adding a `#![recursion_limit="256"]` attribute to your crate (`magic_robot`)
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

If I import one-by-one, it works OK. Am I doing something wrong? =-)

**Comment by de-vri-es** [comment]
Ah, no, you're not doing anything wrong, but star imports are not allowed to conflict. My guess is that it's not importing the `assert` macro because it conflicts with `assert` from the std prelude. However, when I tried such a glob import I got an error about the import itself (if I remember correctly).

Also, why this would cause too deep (probably infinite) recursion is unclear to me. `check` should work fine without importing `assert`. I will need to investigate this in more detail. Thank you for reporting the issue!

**Comment by de-vri-es** [comment]
Ah, it's because we have a custom `stringify` macro for stable and beta that tries to treat the input as expression before falling back to stringifying as arbitrary tokens.

That macro itself just delegated to `stringify!()`, but without using full qualification. Changing it to explicitly use `::core::stringify` helps, but I think we should also avoid leaking that into scope in the first place when you do `use assert2::*`.

**Comment by m-ou-se** [comment]
Something like

```rust
pub use core::stringify as __assert2_core_stringify;
```

could work. And then using it as `$crate::__assert2_core_stringify`.

**Comment by de-vri-es** [comment]
Why wouldn't we just use `::core::stringy` though?

But what I mean is also that I don't want `use assert2::*` to import our `stringify!()`, even if the std prelude is disabled. I'm now moving all pub doc-hidden items to a `_assert2_impl`  module, but the macro is always exported at the crate root, so I'm renaming it to `__assert2_stringify!`.

**Comment by m-ou-se** [comment]
*Technically* one could give `core` a different name by using the unstable `#![no_core]`. ;)

**Comment by de-vri-es** [comment]
Alright, I suppose if we have the tools to do it 100% correct, why would we not :p

**Comment by de-vri-es** [comment]
I'll still move the rest of the doc-hidden things to a safer place, then the fix can be released :)

**Comment by m-ou-se** [comment]
Can't wait for macros 2.0 to be finished and stable. ^^ Then this can all be done properly.

**Comment by de-vri-es** [comment]
Just released 0.3.4 with the fix. All other doc-hidden items now also start with `__assert2`, to avoid name pollution.

@m-ou-se: thanks for the fix, and @StragaSevera: thank you for reporting the issue :)
